### ISSUE TITLE
Uncaught Invariant Violation: unmountComponentAtNode(...): Target container is not a DOM element.

### ISSUE THREAD

**Issue by andrewBalekha** [Hacker0x01/react-datepicker#357]
I'm displaying datepicker depending on a state, and when datepciker should disappear, console throwing error.

`Uncaught Invariant Violation: unmountComponentAtNode(...): Target container is not a DOM element.`

**Comment by rafeememon** [comment]
Thanks for the report.  I reported this issue at souporserious/react-tether#8

**Comment by andrewBalekha** [comment]
:+1:

**Comment by rafeememon** [comment]
Added a failing unit test which will pass when the react-tether bug is fixed.

**Comment by rafeememon** [comment]
@andrewBalekha, we merged a temporary fix for the problem and published v0.23, please give that a try.

(Note the breaking change re: localization if that applies to you, more info in the readme)

**Comment by Atenna** [comment]
I had the same problem yesterday. I am now using v.0.23, no more unmount error.  Thanx for B-day present :+1:

**Comment by andrewBalekha** [comment]
@rafeememon Yep, thanks. works now

**Comment by rafeememon** [comment]
Made a proper fix for this and released v0.23.1

**Comment by vl-lapikov** [comment]
v0.23.1:
Datepicker inside react-bootstrap modal. When I close modal, I get error: unmountComponentAtNode ...

v0.23.0:
Works fine

**Comment by rafeememon** [comment]
@vl-lapikov make sure react-tether is bumped to v0.3.3, maybe try clearing out node_modules and re-installing

**Comment by vl-lapikov** [comment]
@rafeememon didn't help. I made sure react-tether is v0.3.3.

**Comment by rafeememon** [comment]
@vl-lapikov, does it work outside of a react-bootstrap modal?  I'm not familiar with the setup of react-bootstrap or how the modal might make things different

**Comment by vl-lapikov** [comment]
@rafeememon Checked outside react-bootstrap modal and it throws the same error. My steps are about:
- render list of datepickers
- choose date in one of them (datepicker A)
- rerender list of datepickers without datepicker A

I am busy today, however tomorrow might be able to provide short example.

**Comment by TonyVeigel** [comment]
I also upgraded and get this same error, react-tether is at v0.3.3. started with release 2.2

**Comment by rafeememon** [comment]
An example would be great here, ideally it's one of the following

* A simple project that we can run showing the problem
* A failing unit test

**Comment by rafeememon** [comment]
I think this might be an issue with how v0.23.1 was published; there's a similar confusing problem on #352 

cc @martijnrusschen

**Comment by martijnrusschen** [comment]
Can you try this again with v0.23.2?

**Comment by taxigy** [comment]
Having the same problem in 0.23.1. The component hierarchy is pretty basic, still, when I try to unmount parent component and mount another (via react-router), I get the

```
Uncaught Invariant Violation: unmountComponentAtNode(...): Target container is not a DOM element.
```

exception.

**Comment by TonyVeigel** [comment]
Works for me with 23.2. Thanks

**Comment by rafeememon** [comment]
:+1:

**Comment by martijnrusschen** [comment]
@rafeememon did you see https://github.com/Hacker0x01/react-datepicker/issues/357#issuecomment-184847108?

**Comment by rafeememon** [comment]
Yeah, @taxigy was using 0.23.1.  @TonyVeigel confirmed that 0.23.2 works

**Comment by timaschew** [comment]
Can confirm the fix in 0.23.2 as well, thank you!

**Comment by james4388** [comment]
I use inside react router and has this problem too.
